codalab/worker/worker_run_state.py

```diff
@@ -354,7 +354,9 @@ def mount_dependency(dependency, shared_file_system):
                             parent_path=os.path.join(dependency_path, child),
                         )
                     )
-                    self.paths_to_remove.append(child_path)
+                    run_state = run_state._replace(
+                        paths_to_remove=(run_state.paths_to_remove or []) + [child_path]
```

Comment:
Included `or []` so we don't break existing runs on prod (which may not have `paths_to_remove` defined yet).